src/libraries/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs

```diff
@@ -107,6 +107,98 @@ public void CheckEndReachableDoesNotCrossPastEnd()
 
         #endregion
 
+        #region Offset
+
+        [Fact]
+        public void GetOffset_SingleSegment()
+        {
+            var buffer = new ReadOnlySequence<T>(new T[50]);
+            Assert.Equal(25, buffer.GetOffset(buffer.GetPosition(25)));
+        }
+
+        private (BufferSegment<T> bufferSegment1, BufferSegment<T> bufferSegment4) GetBufferSegment()
+        {
+            var bufferSegment1 = new BufferSegment<T>(new T[50]);
+            BufferSegment<T> bufferSegment2 = bufferSegment1.Append(new T[50]);
+            BufferSegment<T> bufferSegment3 = bufferSegment2.Append(new T[0]);
+            return (bufferSegment1, bufferSegment3.Append(new T[50]));
+        }
+
+        private ReadOnlySequence<T> GetFourSegmentsReadOnlySequence()
+        {
+            (BufferSegment<T> bufferSegment1, BufferSegment<T> bufferSegment4) = GetBufferSegment();
+            return new ReadOnlySequence<T>(bufferSegment1, 0, bufferSegment4, 50);
+        }
+
+        [Fact]
+        public void GetOffset_MultiSegment_FirstSegment()
+        {
+            ReadOnlySequence<T> buffer = GetFourSegmentsReadOnlySequence();
+            Assert.Equal(25, buffer.GetOffset(buffer.GetPosition(25)));
+        }
+
+        [Fact]
+        public void GetOffset_MultiSegment_LastSegment()
+        {
+            ReadOnlySequence<T> buffer = GetFourSegmentsReadOnlySequence();
+            Assert.Equal(125, buffer.GetOffset(buffer.GetPosition(125)));
+        }
+
+        [Fact]
+        public void GetOffset_MultiSegment_MiddleSegment()
```

Comment:
How about tests for boundary conditions and also one for empty segment?

For example, here is what your sequence looks like: [50] -> [50] -> [0] -> [50]

What sequence position do I need to pass to get back the offset 50 and 51 (boundary condition between non-empty segments), along with 100 and 101 (boundary condition across an empty segment)? A test with that would be good.

Another one could be: [0] -> [0] -> [0] -> [50]
If I pass all the various valid SequencePositions, we should get back offset 0 for all of them.

Reply:
> If I pass all the various valid SequencePositions, we should get back offset 0 for all of them.

@ahsonkhan this statement it's not clear to me...what do you mean for "valid SequencePositions"?If I pass a position got with offset 1 I'll get offset 1, GetPosition/GetOffset are specular if I understood well.

Reply:
> If I pass a position got with offset 1 I'll get offset 1, GetPosition/GetOffset are specular if I understood well.

Yes, you are right.

Sorry, let me try to clarify. What I meant was the following. Let's say these are the segments of the readonlysequence:
segment1 = [0]
segment2 = [0]
segment3 = [0]
segment4 = [50]

All of these should return 0:
GetOffset(new SequencePosition(segment1, 0)) // passing in anything non-zero should throw
GetOffset(new SequencePosition(segment2, 0)) // passing in anything non-zero should throw
GetOffset(new SequencePosition(segment3, 0)) // passing in anything non-zero should throw
GetOffset(new SequencePosition(segment4, 0))

All of these should return i for i = 0 to 50:
GetOffset(new SequencePosition(segment4, i))

Reply:
Ok now it's clear sorry for my maybe silly question 😅

Reply:
added